// editor.js/Gruntfile.js
/**
 *
 */

module.exports = function(grunt) {

    var timestamp = new Date().getTime() + "";

    // 1. All configuration goes here
    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        watch: {
            scripts: {
                files: ['src/js/*.js'],
                tasks: ['js-task']
            },
            css: {
                files: ['src/css/*.css'],
                tasks: ['css-task']
            }
        },
        concat: {
            options: {
                separator: ';'
            },
            dist: {
                src: ['src/js/*.js'],
                dest: 'dist/editor.js'
            }
        },
        uglify: {
            options: {
                mangle: true,
                preserveComments: 'some',
                banner: '/* <%= pkg.name %> - v<%= pkg.version %>*/\n'
            },
            my_target: {
                files: {
                    'dist/editor.js' : ['src/js/*.js']
                }
            }
        },
        cssmin: {
            options: {
              shorthandCompacting: false
           },
           plugin: {
              files: {
                 'dist/editor.min.css': ['src/css/editor.css']
              }
           }
        }
    });
    // 3. Where we tell Grunt we plan to use this plug-in.
    grunt.loadNpmTasks('grunt-contrib-watch');
    grunt.loadNpmTasks('grunt-contrib-cssmin');
    grunt.loadNpmTasks('grunt-contrib-uglify');
    grunt.loadNpmTasks('grunt-contrib-concat');
    require('time-grunt')(grunt);
    // 4. Where we tell Grunt what to do when we type "grunt" into the terminal.
    grunt.registerTask('default', ['concat', 'cssmin']);
    grunt.registerTask('js-task', ['concat']);
    grunt.registerTask('css-task', ['cssmin']);
};
